cmake/compiler/gcc/target_arm.cmake

```diff
@@ -10,6 +10,11 @@ endif()
 
 list(APPEND ARM_C_FLAGS -mabi=aapcs)
 
+if(CONFIG_BIG_ENDIAN)
```

Comment:
currently `BIG_ENDIAN` is a promptless symbol, meaning it depends on something having a `select BIG_ENDIAN` (or a change of default).

Currently the arch for which that is done is `SPARC`:
https://github.com/zephyrproject-rtos/zephyr/blob/2457f730f43079d8d0626f6dee6f51765a29605d/arch/Kconfig#L65
https://github.com/zephyrproject-rtos/zephyr/blob/2457f730f43079d8d0626f6dee6f51765a29605d/arch/Kconfig#L70

so how is this expected to be selected for `arm v7-r` and thus selecting this code path ?